lib/ansible/plugins/strategy/linear.py

```diff
@@ -401,9 +399,8 @@ def run(self, iterator, play_context):
                 failed_hosts = []
                 unreachable_hosts = []
                 for res in results:
-                    # execute_meta() does not set 'failed' in the TaskResult
-                    # so we skip checking it with the meta tasks and look just at the iterator
-                    if (res.is_failed() or res._task.action == 'meta') and iterator.is_failed(res._host):
+                    # some tasks don't set 'failed' in the result so we can't test 'res.is_failed()'
+                    if (res.is_failed() or res._task.action in ('meta', 'include_role')) and iterator.is_failed(res._host):
```

Comment:
I believe we need to add `include_tasks` here as well.  I wouldn't worry about adding old "include".

Reply:
Tested with `include_tasks` and it is indeed the case. Fixed.